pkg/registry/flowcontrol/ensurer/strategy.go

```diff
@@ -298,6 +298,10 @@ func removeConfiguration(wrapper configurationWrapper, name string) error {
 // Any object present in current is a candidate for removal if both a and b are true:
 //  a. the object in current is missing from the bootstrap configuration
 //  b. the object has the designated auto-update annotation key
+//
+// 'removeConfiguration' checks the value of the annotation key before it makes
```

Comment:
This comment belongs on removeConfiguration, not getRemoveCandidate.
The comment on getRemoveCandidate should say that what is returned is a list of names of objects to _conditionally_ remove.

BTW, why is there a separation between getRemoveCandidate and removeConfiguration?  Why not do it all in one method?

BTW, getRemoveCandidate is named singular and returns plural.

Reply:
The name "removeConfiguration" is unconditional but the behavior is conditional.

Why is the test of the auto-update annotation in removeConfiguration rather than in getRemoveCandidate(s)?